src/Microsoft.Azure.SignalR.AspNet/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -46,8 +47,13 @@ public ServiceEndpointProvider(ServiceEndpoint endpoint, ServiceOptions options)
             _port = endpoint.Port;
             _algorithm = options.AccessTokenAlgorithm;
             Proxy = options.Proxy;
-        }
 
+            if (!_accessKey.InitializedTask.IsCompleted)
```

Comment:
Checking `IsCompleted` sounds a bit tricky to me. we can get if AAD auth is necessary from options?